servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultHttpLoadBalancerFactory.java

```diff
@@ -80,6 +113,35 @@ public FilterableStreamingHttpLoadBalancedConnection toLoadBalancedConnection(
         return new DefaultFilterableStreamingHttpLoadBalancedConnection(connection);
     }
 
+    @Override
+    public FilterableStreamingHttpLoadBalancedConnection toLoadBalancedConnection(
+            final FilterableStreamingHttpConnection connection,
+            final ReservableRequestConcurrencyController concurrencyController,
+            @Nullable final ContextMap context) {
+
+        RequestTracker hostHealthIndicator = null;
+        if (context == null) {
+            LOGGER.debug("Context is null. In order for " + DefaultHttpLoadBalancerFactory.class.getSimpleName() +
+                    ":toLoadBalancedConnection to get access to the " + RequestTracker.class.getSimpleName() +
+                    ", health-monitor of this connection, the context must not be null.");
+        } else {
+            hostHealthIndicator = context.get(REQUEST_TRACKER_KEY);
+            if (hostHealthIndicator == null) {
+                LOGGER.debug(REQUEST_TRACKER_KEY.name() + " is not set in context. " +
+                        "In order for " + DefaultHttpLoadBalancerFactory.class.getSimpleName() +
+                        ":toLoadBalancedConnection to get access to the " + RequestTracker.class.getSimpleName() +
+                        ", health-monitor of this connection, the context must be properly wired.");
+            }
+        }
+
+        if (hostHealthIndicator == null) {
+            return HttpLoadBalancerFactory.super.toLoadBalancedConnection(connection, concurrencyController, context);
```

Comment:
Here we need to call `new DefaultFilterableStreamingHttpLoadBalancedConnection(connection, concurrencyController)`. The super impl has more code for backward compatibility in case users have their own `HttpLoadBalancerFactory`, but here we are in final implementation.